### ISSUE TITLE
Проблемы при подключении с первым каналом если пропускная способность сети низкая

### ISSUE THREAD

**Issue by OmlineEditor** [vladpen/cams#47]
Я далеко и пропускная способность сети низкая, открываю камеру где прошлый раз был выбран **_первый канал_**, на экране только круг со стрелкой что пытается подключиться и **_нет кнопки_** переключения канала, я не могу перевести камеру на другой канал чтобы что-то посмотреть при низкой скорости сети.

Как будем решать проблему? Что делать людям если они попали в такую западню?

**Comment by vladpen** [comment]
Предлагаю просто всегда показывать кнопку во время первой загрузки.

**Comment by OmlineEditor** [comment]
А если это не первая загрузка, а вторая. я перешла за 2 минуты в дальнюю комнату и ухудшился прием и я перезапускаю программу и все засисло т.к. 2 минуты назад я смотрела 1й канал при хорошем сигнале и эти настройки запомнились программой.

**Comment by vladpen** [comment]
Я имел в виду вход на экран камеры. Если программу перезапустить, вход снова будет первым.

**Comment by OmlineEditor** [comment]
Программа может определять есть картинка с видео или нет из-за проблем сети?
Может при проблемах сети всегда переходить при запуске на 2й канал?

**Comment by vladpen** [comment]
Теоретически можно измерять время загрузки буфера и если буфер не заполнен за какое-то пороговое время, пытаться переключиться. Но смысла в этом я не вижу - на моем оборудовании эта проблема не воспроизводится: вайфай либо совсем отваливается, либо ключевой кадр всегда вытягивается (картинка может дергаться, но она есть).

**Comment by OmlineEditor** [comment]
Пока оставим проблему как есть, потом может что-то придумаем или кто-то подскажет решение

**Comment by vladpen** [comment]
Так может кнопку всё-таки показать для переключения вручную?

**Comment by OmlineEditor** [comment]
Да, думаю кнопку переключени канала при открытии камеры нужно показывать всегда, даже тогда когда еще ничего не загрузилось и на экране пусто.
Пусть будет пока кнопка, потом возможно придумаем новый способ.

**Comment by vladpen** [comment]
Выполнено. Если выбран первый канал, можно переключиться на второй не дожидаясь загрузки.
